Userland/Services/SystemServer/main.cpp

```diff
@@ -363,6 +363,11 @@ static void prepare_synthetic_filesystems()
         VERIFY_NOT_REACHED();
     }
 
+    rc = symlink("/proc/self/tty", "/dev/tty");
```

Comment:
Wait. wouldn't this only impact SystemServer's view of the proc filesystem?

Reply:
This is a symlink to /proc/self/tty, so it should point to the reader's proc file system I think

Reply:
That.. works? It seems unusual that userspace, even the root user, can create a symlink from "x" to "/proc/self/something" and have it show up in every process's /proc/self

Reply:
I think one of us is misunderstanding something, because at least as far as I understand `proc/self/tty` already exists, we are creating `dev/tty` which is a symlink to the `proc/self/tty`

Reply:
`/proc/self/tty` is the *pointed-to* file of the symbolic link. `/dev/stdin`, `/dev/stdout` and `/dev/stderr` are set up the same way. systemd sets up symlinks the same way.

Reply:
aha! It's me :sweat_smile: . Got the link target and link path swapped in my head.